From 5ef23041f1944d013a91b804ac5dabe7f753d407 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Mon, 4 Mar 2024 09:49:11 +0100
Subject: [PATCH 0975/1141] arm64: dts: st: remove cs42l51 codec support on
 stm32mp157f-dk

Remove Cirrus CS42L51 codec from audio sound card on STM32MP257F
DK board, as this audio codec is removed on DK REVC board.

Change-Id: I2de2407cef87369a1183446e526fe3ef2285e4d4
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/364295
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 80 +----------------------
 1 file changed, 1 insertion(+), 79 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -165,11 +165,7 @@
 	sound {
 		compatible = "audio-graph-card";
 		label = "STM32MP25-DK";
-		routing =
-			"Playback" , "MCLK",
-			"Capture" , "MCLK",
-			"MICL" , "Mic Bias";
-		dais = <&sai1a_port &sai1b_port &i2s2_port &mdf1_port0 &mdf1_port1>;
+		dais = <&i2s2_port &mdf1_port0 &mdf1_port1>;
 		status = "okay";
 	};
 
@@ -380,39 +376,6 @@
 		reset-gpios = <&gpioi 0 GPIO_ACTIVE_LOW>;
 		status = "okay";
 	};
-
-	cs42l51: cs42l51@4a {
-		compatible = "cirrus,cs42l51";
-		reg = <0x4a>;
-		#sound-dai-cells = <0>;
-		VL-supply = <&scmi_v3v3>;
-		VD-supply = <&scmi_v1v8>;
-		VA-supply = <&scmi_v1v8>;
-		VAHP-supply = <&scmi_v1v8>;
-		reset-gpios = <&gpiod 6 GPIO_ACTIVE_LOW>;
-		clocks = <&sai1a>;
-		clock-names = "MCLK";
-		status = "okay";
-
-		cs42l51_port: port {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			cs42l51_tx_endpoint: endpoint@0 {
-				reg = <0>;
-				remote-endpoint = <&sai1a_endpoint>;
-				frame-master = <&cs42l51_tx_endpoint>;
-				bitclock-master = <&cs42l51_tx_endpoint>;
-			};
-
-			cs42l51_rx_endpoint: endpoint@1 {
-				reg = <1>;
-				remote-endpoint = <&sai1b_endpoint>;
-				frame-master = <&cs42l51_rx_endpoint>;
-				bitclock-master = <&cs42l51_rx_endpoint>;
-			};
-		};
-	};
 };
 
 &i2c8 {
@@ -593,47 +556,6 @@
 	status = "okay";
 };
 
-&sai1 {
-	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&sai1a_pins_a>, <&sai1b_pins_a>;
-	pinctrl-1 = <&sai1a_sleep_pins_a>, <&sai1b_sleep_pins_a>;
-	status = "okay";
-
-	sai1a: audio-controller@40290004 {
-		#clock-cells = <0>;
-		dma-names = "tx";
-		status = "okay";
-
-		sai1a_port: port {
-			sai1a_endpoint: endpoint {
-				remote-endpoint = <&cs42l51_tx_endpoint>;
-				format = "i2s";
-				mclk-fs = <256>;
-				dai-tdm-slot-num = <2>;
-				dai-tdm-slot-width = <32>;
-			};
-		};
-	};
-
-	sai1b: audio-controller@40290024 {
-		dma-names = "rx";
-		st,sync = <&sai1a 2>;
-		clocks = <&rcc CK_KER_SAI1>, <&sai1a>;
-		clock-names = "sai_ck", "MCLK";
-		status = "okay";
-
-		sai1b_port: port {
-			sai1b_endpoint: endpoint {
-				remote-endpoint = <&cs42l51_rx_endpoint>;
-				format = "i2s";
-				mclk-fs = <256>;
-				dai-tdm-slot-num = <2>;
-				dai-tdm-slot-width = <32>;
-			};
-		};
-	};
-};
-
 &scmi_regu {
 	scmi_vddio1: regulator@0 {
 		regulator-min-microvolt = <1800000>;
